client/src/components/SigninButton.svelte

```diff
@@ -1,19 +1,151 @@
 <script lang="ts">
   import Modal from "./Modal.svelte";
   import { authStore as user } from "$stores/auth";
+  import { loginWithGoogle, loginWithMicrosoft, loginWithEmail, onSignOut, createUser } from "$lib/firebase/auth";
 
   // check if the user is logged in with getAuth
   let openSignInModal = false;
+  let email = "";
+  let password = "";
+  let confirmPass = "";
+  let clickedButton = false;
+  let clickedButton2 = false;
+  let errorMessage = "";
+  async function createAccount() {
+    if (password == "") {
+      return;
+    }
+    if (password != confirmPass) {
+      errorMessage = "Password does not match";
+      password = "";
+      confirmPass = "";
+      return;
+    }
+    if (email == "") {
+      return;
+    } else {
+      try {
+        await createUser(email, password);
+        errorMessage = "";
+      } catch (err) {
+        errorMessage = "Invalid Username or Password";
+      }
```

Comment:
not all errors are a result of an Invalid Username or Password you should check the `err` object and display a relivant message to the user

Reply:
resolved